src/components/SpecialResolution/SpecialResolutionForm.vue

```diff
@@ -376,10 +389,30 @@ export default class SpecialResolutionForm extends Mixins(DateMixin) {
       ...this.getcreateResolution,
       signingPerson: this.signingPerson
     })
+    await this.$nextTick()
+    this.validate()
   }
-  /** Set values if exist */
+   /** called to store component validity to store. */
+   protected async validate () {
+     // wait to reflect validation state
+     await this.$nextTick()
```

Comment:
This will do nothing unless you await this method.

Reply:
When I am removing validation is not working properly. 
I could not figure it out exactly. Am I missing something? 
with 2 await
[not-working2.webm](https://user-images.githubusercontent.com/55506249/184027210-37271f12-ef40-472b-9679-759adbf390ae.webm)
with one await
[not-working1.webm](https://user-images.githubusercontent.com/55506249/184027214-02741736-bf29-4702-9825-e2ba1d9d301b.webm)
with one await
[working1.webm](https://user-images.githubusercontent.com/55506249/184027184-19ed8e81-9bb4-4f8c-935a-a92d03505c3d.webm)

Reply:
Yes, you are missing that you need to `await this.validate()` otherwise the inner async stuff won't work.

Reply:
it didn't work for me :(
[notworking3.webm](https://user-images.githubusercontent.com/55506249/184029911-7179c1de-5e2f-408a-9c0f-a9ca6973aecf.webm)

Reply:
This is the code in your video:

![image](https://user-images.githubusercontent.com/10002889/184030586-e67c65e0-87ad-46a7-977e-f3ad20a05e07.png)

You are still not awaiting the validate method.

Try this:
line 387: delete it
line 388: `await this.validate()`

Reply:
with this code also not working
```
   @Watch('signatory', { deep: true })
  protected async onsignatoryChanged (): Promise<void> {
    this.setResolution({
      ...this.getcreateResolution,
      signatory: this.signatory
    })

    await this.validate()
  }
```
[notworking4.webm](https://user-images.githubusercontent.com/55506249/184032176-ee8277eb-7398-424d-9894-8f87a4eb7388.webm)

Reply:
I found the validation method in this component works like a charm: [link](https://github.com/bcgov/business-edit-ui/blob/main/src/components/common/YourCompany/CompanyName/CorrectNameRequest.vue). Hope this can give a little help.

Reply:
In this link, we have all values in local property and form validation is emitted to different components. 
But in our case, there is no submit/done button or we dont have any parent to emit. we need to save component status into store. SO we cant use like link :(

Reply:
**Re: await issue**

Faiz and I looked at this together and tested it. We found that, if validate() is not awaited, it returns as soon as it hits its "await nextTick", and the statement after its call executes. This is risky code if there is a statement after the call to validate(), since validate() hasn't finished running. To resolve this, all calls to validate() are awaited (even though they cause more work for the framework and execution).

**Re: multiple nextTick issue**

A nextTick() is needed for the store action (set) to complete, and a second nextTick() is needed for the component validations to complete. The second nextTick already had a comment explaining it. A comment was added to the first nextTick to explain it.

I think this is a bit weird but (a) it's apparently needed for the reasons stated, and (b) this way each method handles what it needs to know about -- specifically, a method doesn't need to worry about another method's concerns.

 I have no further concerns with these issues and am resolving this conversation.